### PR TITLE
Add defaulted impls for Send and Sync

### LINKED ISSUE #23225: impl Trait for .. {} does not require a feature gate
I suspect that we may want more experience with this syntax before freezing it, but I may also be misremembering!

cc @nikomatsakis 
cc @FlaPer87

### PR DESCRIPTION
**Pull request by flaper87**
Fixes #23225

r? @nikomatsakis

### CODE REVIEW

**Comment by nikomatsakis** [comment]
@FlaPer87 rebase?

**Comment by FlaPer87** [comment]
@nikomatsakis r?

File: src/libcore/marker.rs [hunk 1]
```diff
@@ -39,6 +39,8 @@ pub unsafe trait Send : MarkerTrait {
     // empty.
 }
 
+impl Send for .. { }
```
Comment (FlaPer87) [inline src/libcore/marker.rs]: this should be `unsafe`

**Comment by nikomatsakis** [comment]
r+ modulo suggested refactoring

**Comment by Manishearth** [comment]
```
foo.rs:16:5: 16:23 error: default trait implementations are experimental and possibly buggy
foo.rs:16     impl Bar for .. {}
              ^~~~~~~~~~~~~~~~~~
foo.rs:16:23: 16:23 help: add #![feature(optin_builtin_traits)] to the crate attributes to enable
error: aborting due to previous error
make[1]: *** [all] Error 101

```

**Pull request merged**